src/chatd.cpp

```diff
@@ -4735,6 +4879,10 @@ void Chat::msgIncomingAfterDecrypt(bool isNew, bool isLocal, Message& msg, Idx i
             mAttachmentNodes->addMessage(msg, isNew, false);
         }
         CALL_DB(addMsgToHistory, msg, idx);
+        if (!isNew || mOldestIdxInDb == CHATD_IDX_INVALID)
+        {
+            mOldestIdxInDb = mDbInterface->getOldestIdx();
```

Comment:
Again, I think here you should be able to find the oldest idx without the need of the DB query, right?